### ISSUE TITLE
[Circle]: Onboarding/connection workflow

### ISSUE THREAD

**Issue by bragefuglseth** [SoongNoonien/plattenalbum#131]
Breakout thread from the ongoing [GNOME Circle review](https://gitlab.gnome.org/Teams/Circle/-/issues/219) of the app.

When opening the app for the first time, it should be obvious what the "next step" is even if you don't have any previous knowledge of MPD. There is also no feedback if the connection fails, the button just doesn't do anything. I managed to work out that I needed to open the Help page from the main menu, but ideally people shouldn't have to figure that out on their own. They should just be guided into it, for example like this:

![connection](https://github.com/user-attachments/assets/19a9e7ee-5fa7-4c53-a75e-d2fbfb1b28ee)

This can also be used at later stages in the app; instead of connecting to e.g. a remote database by tweaking stuff in Settings, there can be a "disconnect" item in the main menu of the app that takes you back to this page.

**Comment by SoongNoonien** [comment]
Ah, another question. Should this page also be shown if the connection to MPD is lost while simply using the client and not explicitly pressing the "disconnect" button?

**Comment by bragefuglseth** [comment]
> most of 1st comment

Change it however you see fit 😁 I was thinking that if someone wants to save the config file another way, they could just select the text manually and copy it, but do what you think makes the most sense.

> Also if we provide distribution specific installation instructions we should also provide distribution specific "initialization" instructions. For example in Ubuntu the server is started with the installation in root context.

This sounds like a good idea to provide a couple of distro-specific commands for even if we don’t do it for the installation. It’s less likely to change over time than the installation method, and people are much less likely to just «know» the right command for it.

> Ah, another question. Should this page also be shown if the connection to MPD is lost while simply using the client and not explicitly pressing the "disconnect" button?

Sounds like a good idea! Just make sure to indicate clearly that the connection was lost, e.g. with an AdwToast and (if the app is out of focus) a notification.

**Comment by SoongNoonien** [comment]
I've addressed this in master now. I've deviated from the initial proposal a bit. After checking ubuntu again I've noticed that using the provided systemd command is actually correct. This command will set up a instance in the user context which has higher priority when connecting in Plattenalbum. So I don't see the need to provide distribution specific instructions anymore. I think systemd can safely be assumed to be present, if it is not then the respective users probably know what they are doing anyway.
